crates/bevy_core/src/time/time.rs

```diff
@@ -34,58 +49,171 @@ impl Time {
     pub(crate) fn update_with_instant(&mut self, instant: Instant) {
         if let Some(last_update) = self.last_update {
             self.delta = instant - last_update;
-            self.delta_seconds_f64 = self.delta.as_secs_f64();
             self.delta_seconds = self.delta.as_secs_f32();
+            self.delta_seconds_f64 = self.delta.as_secs_f64();
+            self.time_since_startup += self.delta.mul_f32(self.relative_speed);
+        } else {
+            self.first_update = Some(instant);
+            self.time_since_startup += (instant - self.startup).mul_f32(self.relative_speed);
```

Comment:
Should this really be multiplied by `relative_speed`? This is a starting point, isn't it? And why is the time incremented instead of overwritten?

Reply:
This field is incremented because it's affected by changes to `relative_speed`. One of the "real" versions will give you the absolute duration.

Reply:
I see. But still, we agree we're in the case of the first update, so why increment `self.time_since_startup` instead of overwriting it? Shouldn't it be:
```suggestion
            self.time_since_startup = (instant - self.startup).mul_f32(self.relative_speed);
```
?

Reply:
It's incremented because `relative speed` affects the rate time advances relative to CPU time. We're not literally scaling time, only delta time.

Also, `relative_speed` can change freely, so just overwriting the value would lead to time going backwards (e.g. set `relative_speed` to 4x, then 2x). It has to be monotonically increasing.